File: src/main/java/project/rpg/effect/Stun.java
package project.rpg.effect;

import org.bukkit.entity.LivingEntity;
import org.bukkit.potion.PotionEffect;
import org.bukkit.potion.PotionEffectType;

public class Stun extends EffectBase {

    @Override
    public void effect() {
        this._entity.addPotionEffect(new PotionEffect(PotionEffectType.SLOW, 20*_second, 7,false,false,false));
    }

    public Stun(LivingEntity entity, int second) {
        super(entity, second);
    }
    public Stun(LivingEntity entity,long tick) {
        super(entity,tick);
    }

}

File: src/main/kotlin/project/rpg/effects/objects/Stun.kt
package project.rpg.effects.objects

import org.bukkit.entity.LivingEntity
import org.bukkit.potion.PotionEffect
import org.bukkit.potion.PotionEffectType
import project.rpg.effects.EffectBase

class Stun : EffectBase {
    override fun effect(durationMillis: Long) {
        entity.addPotionEffect(PotionEffect(PotionEffectType.SLOW, 20 * second, 7, false, false, false))
    }

    constructor(entity: LivingEntity, second: Int) : super(entity, second)
    constructor(entity: LivingEntity, tick: Long) : super(entity, tick)
}
